test/new-e2e/tests/agent-metric-logs/linux-log-agent/01_file_tailing_test.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2016-present Datadog, Inc.
+
+package logagent
+
+import (
+	_ "embed"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+
+	e2e "github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+)
+
+// LinuxVMFakeintakeSuite defines a test suite for the log agent interacting with a virtual machine and fake intake.
+type LinuxVMFakeintakeSuite struct {
+	e2e.Suite[e2e.FakeIntakeEnv]
+	DevMode bool
+}
+
+//go:embed log-config/log-config.yaml
+var logConfig []byte
+
+// logsExampleStackDef returns the stack definition required for the log agent test suite.
+func logsExampleStackDef() *e2e.StackDefinition[e2e.FakeIntakeEnv] {
+
+	return e2e.FakeIntakeStackDef(
+		e2e.WithAgentParams(
+			agentparams.WithLogs(),
+			agentparams.WithIntegration("custom_logs.d", string(logConfig))))
+
+}
+
+// TestE2EVMFakeintakeSuite runs the E2E test suite for the log agent with a VM and fake intake.
+func TestE2EVMFakeintakeSuite(t *testing.T) {
+	s := &LinuxVMFakeintakeSuite{}
+	_, s.DevMode = os.LookupEnv("TESTS_E2E_DEVMODE")
+
+	e2e.Run(t, s, logsExampleStackDef())
+}
+
+func (s *LinuxVMFakeintakeSuite) BeforeTest(_, _ string) {
+	// Flush server and reset aggregators before the test is ran
+	s.cleanUp()
+}
+
+func (s *LinuxVMFakeintakeSuite) TearDownSuite() {
+	// Flush server and reset aggregators after the test is ran
+	s.cleanUp()
+}
+
+func (s *LinuxVMFakeintakeSuite) TestLinuxLogTailing() {
+	// Run test cases
+	s.Run("LogCollection", s.LogCollection)
+
+	s.Run("LogPermission", s.LogPermission)
+
+	s.Run("LogRotation", s.LogRotation)
+}
+
+func (s *LinuxVMFakeintakeSuite) LogCollection() {
+	t := s.T()
+	fakeintake := s.Env().Fakeintake
+
+	// Create a new log file
+	s.Env().VM.Execute("sudo touch /var/log/hello-world.log")
+
+	// Part 1: Ensure no logs are present in fakeintake
+	s.EventuallyWithT(func(c *assert.CollectT) {
+		logs, err := fakeintake.FilterLogs("hello")
+		if !assert.NoError(c, err, "Unable to filter logs by the service 'hello'.") {
+			return
+		}
+		// If logs are found, print their content for debugging
+		if !assert.Empty(c, logs, "Logs were found when none were expected.") {
+			cat, _ := s.Env().VM.ExecuteWithError("cat /var/log/hello-world.log")
+			t.Logf("Logs detected when none were expected: %v", cat)
+		}
+	}, 5*time.Minute, 10*time.Second)
+
+	// Part 2: Adjust permissions of new log file
+	_, err := s.Env().VM.ExecuteWithError("sudo chmod 777 /var/log/hello-world.log")
+	assert.NoError(t, err, "Unable to adjust permissions for the log file '/var/log/hello-world.log'.")
+
+	// Generate log
+	generateLog(s, "hello-world")
+
+	// Part 3: Assert that logs are found in intake after generation
+	checkLogs(s, "hello", "hello-world")
+}
+
+func (s *LinuxVMFakeintakeSuite) LogPermission() {
+
+	// Part 4: Block permission and check the Agent status
```

Comment:
we technically check for the existence of the log file via error from the chmod and generate log since both will generate errors if the log file does not exist. I can put a sanity check at the beginning of the function to see if the log file exist